mp/src/game/client/c_baseanimating.cpp

```diff
@@ -3179,8 +3179,34 @@ int C_BaseAnimating::DrawModel( int flags )
 		{
 			extraFlags |= STUDIO_IGNORE_NEO_EFFECTS;
 		}
-#endif
 
+		auto pLocalPlayer = C_NEO_Player::GetLocalNEOPlayer();
+		Assert(pLocalPlayer);
+
+		const bool inMotionVision = pLocalPlayer && pLocalPlayer->IsInVision() && pLocalPlayer->GetClass() == NEO_CLASS_ASSAULT;
+		
+		Vector vel;
+		if (IsRagdoll())
+		{
+			vel = GetOldVelocity();
+		}
+		else
+		{
+			vel = GetRootMoveParent()->GetAbsVelocity();
+			if (vel == vec3_origin)
+			{
+				EstimateAbsVelocity(vel);
+			}
+		}
+		bool isMoving = false;
+		if (inMotionVision && vel.LengthSqr() > 0.25 && !IsViewModel() && !(extraFlags & STUDIO_IGNORE_NEO_EFFECTS)) // MOVING_SPEED_MINIMUM ^2
+		{
+			isMoving = true;
+			IMaterial* pass = materials->FindMaterial("dev/motion_third", TEXTURE_GROUP_MODEL);
+			Assert(pass && !IsErrorMaterial(pass));
```

Comment:
Nit: `IsErrorMaterial` already checks for nullptr, so the first `pass` assert is redundant here.